lib/ansible/modules/windows/win_get_url.ps1

```diff
@@ -91,37 +103,202 @@ Function CheckModified-File($module, $url, $dest, $headers, $credentials, $timeo
+        $webFileLength = (Get-AnsibleItem -Path $srcUri.AbsolutePath).Length
+
+        $module.Result.status_code = 200
+        $module.Result.msg = 'OK'
+    } else {
+
+        $webRequest = GetWebRequest -url $url -headers $headers -credentials $credentials -timeout $timeout -use_proxy $use_proxy -proxy $proxy
+        $webRequestPaired = GetWebRequest -url $url -headers $headers -credentials $credentials -timeout $timeout -use_proxy $use_proxy -proxy $proxy
+
+        if ($webRequest -is [System.Net.FtpWebRequest]) {
+            $webRequest.Method = [System.Net.WebRequestMethods+Ftp]::GetDateTimestamp
+        } else {
+            $webRequest.Method = [System.Net.WebRequestMethods+Http]::Head
+        }
+
+        # FIXME: Split both try-statements and single-out catched exceptions with more specific error messages
+        Try {
+            if ($webRequest -is [System.Net.FtpWebRequest]) {
+                $webRequest.Method = [System.Net.WebRequestMethods+Ftp]::GetDateTimestamp
+                $webResponse = $webRequest.GetResponse()
+                $webLastMod = $webResponse.LastModified
+
+                $webRequestPaired.Method = [System.Net.WebRequestMethods+Ftp]::GetFileSize
+                $webResponsePaired = $webRequestPaired.GetResponse()
+                $webFileLength = $webResponsePaired.ContentLength
+            } else {
+                $webRequest.Method = [System.Net.WebRequestMethods+Http]::Head
+                $webResponse = $webRequest.GetResponse()
+                $webLastMod = $webResponse.LastModified
+                $webFileLength = $webResponse.ContentLength
+            }
+        } Catch [System.Net.WebException] {
+            $module.Result.status_code = [int] $_.Exception.Response.StatusCode
+            $module.FailJson("Error requesting '$url'. $($_.Exception.Message)", $_)
+        } Catch {
+            $module.FailJson("Error when requesting 'Last-Modified' date from '$url'. $($_.Exception.Message)", $_)
+        }
+        Finally {
+            # in case GetResponse timeout, $webResponse == $null
+            if($webResponse) {
+               $webResponse.Close()
+            }
+            if($webResponsePaired) {
+                $webResponsePaired.Close()
+             }
+        }
+        $module.Result.status_code = [int] $webResponse.StatusCode
+        $module.Result.msg = [string] $webResponse.StatusDescription
+    }
+
+    if ($check_length) {
+        if ($webFileLength) {
+            if($webFileLength -eq -1){
+                return $false
+            }
+            if($webFileLength -ne $fileLength) {
+                return $true
+            }
+        }
         return $false
     } else {
-        return $true
+        if ($webLastMod -and ((Get-Date -Date $webLastMod).ToUniversalTime() -le $fileLastMod)) {
+            return $false
+        } else {
+            return $true
+        }
+    }
+}
+
+Function Get-FileChecksum {
```

Comment:
I see this is a direct copy from Legacy.psm1 which contains a bit more things we really need. I would copy https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/windows/win_stat.ps1#L16 as it is a simpler implementation and we should only be concerned about files.

Reply:
I think we should create a common function in shared module like it was Legacy.psm1 and start using it and don't duplicate same code in different module.

Reply:
@jborean93, as I can see new version of Get-FileChecksum is not direct copy from Legacy.psm1. It's adjusted to handle most cases of client-side PS versions. To me it looks much better than version from win_stat module you mentioned. Is it necessary to use win_stat version instead of current?

Reply:
It's not a direct copy but everything that's in the Legacy.psm1 file isn't needed for this module. The one in win_stat is a very cutdown version to get the checksum of a file without anything special in there.

Reply:
Fixed